app/schema.graphql

```diff
@@ -0,0 +1,103 @@
+scalar Datetime
+
+type Author {
+  name: String
+  email: String
+}
+
+# git commit data.
+type Commit {
+  sha1: String!
+  author: Author!
+  date: Datetime!
+  message: String!
+  refs: [Ref]
+  objectName: ObjectName!
+}
+
+# file info and metadata.
+type Info {
+  # base name of the file.
+  name: String!
+  # length in bytes.
+  size: Int!
```

Comment:
For a directory is this the `size` of all files beneath it?

Reply:
I'd assume it to be `-1` when `isDIrectory` is true.

Reply:
Why not move this to `Blob` then?

Reply:
A possibility, but info is used in directory listing. We don't use it anywhere yet, would wonder if you'd be interested in the size without the file content.

Reply:
I could see how getting the size of a directory would be valuable for future UX. However at the moment we would only be interested in file sizes.

Reply:
Well I don't see it as being hard to calculate the size of a `Directory` in my model. It's basically `directory.files.map(|size|).sum + directory.sub_directories.map(|size|).sum` :)

Reply:
So if it's useful, it can be added!

Reply:
I think it would be useful!

Reply:
You got it, dear customer!